src/internal/required_argument.py

```diff
@@ -0,0 +1,18 @@
+from internal.argument import Argument
+
+# Class to represent a required argument to a command line tool
+class RequiredArgument(object):
+    def __init__(self, argument=None):
+        self.argument = argument
+
+    # Function to build a RequiredArgument object based on collections objects
+    # Raises KeyError if the argument's name isn't present
```

Comment:
This function doesn't raise a `KeyError` at the moment, but actually we probably want to